src/widget/wcolorpicker.cpp

```diff
@@ -0,0 +1,95 @@
+#include "widget/wcolorpicker.h"
+
+#include <QGridLayout>
+#include <QMapIterator>
+#include <QPushButton>
+#include <QStyle>
+
+#include "util/color/color.h"
+#include "util/parented_ptr.h"
+
+namespace {
+const int kNumColumns = 4;
+}
+
+WColorPicker::WColorPicker(QWidget* parent)
+        : QWidget(parent) {
+    // If another title would be more appropriate in some context, setTitle
+    // can be called again after construction.
+    QGridLayout* pLayout = new QGridLayout();
+    pLayout->setMargin(0);
+    pLayout->setContentsMargins(0, 0, 0, 0);
+
+    m_pStyle = QStyleFactory::create(QString("fusion"));
```

Comment:
Unfortunately, [not all styles supported by Qt support setting a background color for QPushButtons](https://bugreports.qt.io/browse/QTBUG-11089). For example, @ronso0 [was using the `gtk2` style](https://github.com/mixxxdj/mixxx/pull/2362#issuecomment-557915926) and all color buttons just turned out grey. It's possible to work around this by modifing the button border with a qss stylesheet, so that the QStyle will be overwritten (and I guess most skins will do that when they style the WColorPicker), but for skins that don't do that we need to make sure that everyone can see the colors independent of platform.

A style that does support the background color is the platform-independent [Fusion QStyle](https://www.qt.io/blog/2012/10/30/cleaning-up-styles-in-qt5-and-adding-fusion). I think we can safely do that because most skins will customize the buttons anyway (making the style irrelevant). For all others, it's better to have a button that looks slightly different than have buttons without any colors (making the colorpicker unusable).

Should I add a code comment as well?

Reply:
Yes, that would be nice.
Maybe we stumble over it later again.

Reply:
--online-- -> inline

Reply:
Done.